Bug: Compiling android version - argument type 'bool?' can't be assigned to the parameter type 'bool'

used version is the latest one - 5.0.0

I do receive "argument type 'bool?' can't be assigned to the parameter type 'bool'"
```

"Install Android SDK Platform 34 (revision: 2)" finished.
../../.pub-cache/hosted/pub.dev/flutter_platform_widgets-5.0.0/lib/src/platform_text_field.dart:583:30: Error: The argument type 'bool?' can't be assigned to the parameter type 'bool' because 'bool?' is nullable and 'bool' isn't.
      enabled: data?.enabled ?? enabled,
                             ^
Target kernel_snapshot failed: Exception


FAILURE: Build failed with an exception.

* Where:
Script '/Users/builder/programs/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1350

* What went wrong:
Execution failed for task ':app:compileFlutterBuildRelease'.
> Process 'command '/Users/builder/programs/flutter/bin/flutter'' finished with non-zero exit value 1

```

What could be the issue?

--- a/lib/src/platform_app_bar.dart
+++ b/lib/src/platform_app_bar.dart
@@ -321,8 +321,7 @@ class PlatformAppBar
     }
 
     final middleWithMediaQuery = MediaQuery(
-      data: MediaQueryData(
-          textScaler: TextScaler.linear(MediaQuery.textScaleFactorOf(context))),
+      data: MediaQueryData(textScaler: MediaQuery.textScalerOf(context)),
       child: middleLocal,
     );
 

--- a/lib/src/platform_app_bar.dart
+++ b/lib/src/platform_app_bar.dart
@@ -10,10 +10,10 @@ import 'package:flutter/services.dart';
 import 'package:flutter/widgets.dart';
 
 import 'extensions.dart';
+import 'parent_widget_finder.dart';
 import 'platform.dart';
 import 'platform_provider.dart';
 import 'widget_base.dart';
-import 'parent_widget_finder.dart';
 
 //the default has alpha which will cause the content to slide under the header for ios
 const Color _kDefaultNavBarBorderColor = const Color(0x4C000000);
@@ -322,7 +322,7 @@ class PlatformAppBar
 
     final middleWithMediaQuery = MediaQuery(
       data: MediaQueryData(
-          textScaleFactor: MediaQuery.textScaleFactorOf(context)),
+          textScaler: TextScaler.linear(MediaQuery.textScaleFactorOf(context))),
       child: middleLocal,
     );
 

--- a/lib/src/platform_icons.dart
+++ b/lib/src/platform_icons.dart
@@ -188,20 +188,24 @@ class PlatformIcons {
       isMaterial(context) ? Icons.cloud : CupertinoIcons.cloud_fill;
 
   /// Icons: Icons.cloud_download_outlined : CupertinoIcons.cloud_download
-  IconData get cloudDownload =>
-      isMaterial(context) ? Icons.cloud_download_outlined : CupertinoIcons.cloud_download;
+  IconData get cloudDownload => isMaterial(context)
+      ? Icons.cloud_download_outlined
+      : CupertinoIcons.cloud_download;
 
   /// Icons: Icons.cloud_download : CupertinoIcons.cloud_download_fill
-  IconData get cloudDownloadSolid =>
-      isMaterial(context) ? Icons.cloud_download : CupertinoIcons.cloud_download_fill;
+  IconData get cloudDownloadSolid => isMaterial(context)
+      ? Icons.cloud_download
+      : CupertinoIcons.cloud_download_fill;
 
   /// Icons: Icons.cloud_upload_outlined : CupertinoIcons.cloud_upload
-  IconData get cloudUpload =>
-      isMaterial(context) ? Icons.cloud_upload_outlined : CupertinoIcons.cloud_upload;
+  IconData get cloudUpload => isMaterial(context)
+      ? Icons.cloud_upload_outlined
+      : CupertinoIcons.cloud_upload;
 
   /// Icons: Icons.cloud_upload : CupertinoIcons.cloud_upload_fill
-  IconData get cloudUploadSolid =>
-      isMaterial(context) ? Icons.cloud_upload : CupertinoIcons.cloud_upload_fill;
+  IconData get cloudUploadSolid => isMaterial(context)
+      ? Icons.cloud_upload
+      : CupertinoIcons.cloud_upload_fill;
 
   /// Icons: Icons.collections : CupertinoIcons.collections
   IconData get collections =>

--- a/lib/src/platform_text.dart
+++ b/lib/src/platform_text.dart
@@ -32,7 +32,12 @@ class PlatformText extends StatelessWidget {
     TextDirection? textDirection,
     bool? softWrap,
     TextOverflow? overflow,
+    @Deprecated(
+      'Use textScaler instead. '
+      'This feature was removed after version 3.16',
+    )
     double? textScaleFactor,
+    TextScaler? textScaler,
     int? maxLines,
     Locale? locale,
     String? semanticsLabel,
@@ -51,7 +56,7 @@ class PlatformText extends StatelessWidget {
               textDirection: textDirection,
               softWrap: softWrap,
               overflow: overflow,
-              textScaleFactor: textScaleFactor,
+              textScaler: textScaler,
               maxLines: maxLines,
               locale: locale,
               semanticsLabel: semanticsLabel,

--- a/lib/src/platform_text_field.dart
+++ b/lib/src/platform_text_field.dart
@@ -70,7 +70,7 @@ abstract class _BaseData {
     this.onEditingComplete,
     this.onSubmitted,
     this.inputFormatters,
-    this.enabled,
+    this.enabled = true,
     this.cursorWidth,
     this.cursorRadius,
     this.cursorColor,
@@ -139,7 +139,7 @@ abstract class _BaseData {
   final void Function()? onEditingComplete;
   final void Function(String)? onSubmitted;
   final List<TextInputFormatter>? inputFormatters;
-  final bool? enabled;
+  final bool enabled;
   final double? cursorWidth;
   final Radius? cursorRadius;
   final Color? cursorColor;
@@ -348,7 +348,7 @@ class PlatformTextField
   final void Function()? onEditingComplete;
   final void Function(String)? onSubmitted;
   final List<TextInputFormatter>? inputFormatters;
-  final bool? enabled;
+  final bool enabled;
   final double? cursorWidth;
   final Radius? cursorRadius;
   final DragStartBehavior? dragStartBehavior;
@@ -425,7 +425,7 @@ class PlatformTextField
     this.onEditingComplete,
     this.onSubmitted,
     this.inputFormatters,
-    this.enabled,
+    this.enabled = true,
     this.cursorWidth,
     this.cursorRadius,
     this.cursorColor,

--- a/lib/src/platform_theme.dart
+++ b/lib/src/platform_theme.dart
@@ -47,10 +47,10 @@ class _PlatformThemeState extends State<PlatformTheme>
     _themeMode = widget.themeMode;
     _materialLightTheme = _useMaterial3Light == null
         ? widget.materialLightTheme
-        : widget.materialLightTheme?.copyWith(useMaterial3: _useMaterial3Light);
+        : widget.materialLightTheme?.copyWith();
     _materialDarkTheme = _useMaterial3Dark == null
         ? widget.materialDarkTheme
-        : widget.materialDarkTheme?.copyWith(useMaterial3: _useMaterial3Dark);
+        : widget.materialDarkTheme?.copyWith();
     _cupertinoLightTheme = widget.cupertinoLightTheme;
     _cupertinoDarkTheme = widget.cupertinoDarkTheme;
     _matchCupertinoSystemChromeBrightness =
@@ -136,19 +136,15 @@ class _PlatformThemeState extends State<PlatformTheme>
     setState(() {
       if (applyToBothDarkAndLightTheme) {
         _useMaterial3Dark = _useMaterial3Light = useMaterial3;
-        _materialDarkTheme =
-            _materialDarkTheme?.copyWith(useMaterial3: useMaterial3);
-        _materialLightTheme =
-            _materialLightTheme?.copyWith(useMaterial3: useMaterial3);
+        _materialDarkTheme = _materialDarkTheme?.copyWith();
+        _materialLightTheme = _materialLightTheme?.copyWith();
       } else {
         isDark
             ? _useMaterial3Dark = useMaterial3
             : _useMaterial3Light = useMaterial3;
         isDark
-            ? _materialDarkTheme =
-                _materialDarkTheme?.copyWith(useMaterial3: useMaterial3)
-            : _materialLightTheme =
-                _materialLightTheme?.copyWith(useMaterial3: useMaterial3);
+            ? _materialDarkTheme = _materialDarkTheme?.copyWith()
+            : _materialLightTheme = _materialLightTheme?.copyWith();
       }
     });
   }

--- a/lib/src/platform_theme.dart
+++ b/lib/src/platform_theme.dart
@@ -45,21 +45,12 @@ class _PlatformThemeState extends State<PlatformTheme>
 
   void _init() {
     _themeMode = widget.themeMode;
-    _materialLightTheme = _useMaterial3Light == null
-        ? widget.materialLightTheme
-        : widget.materialLightTheme?.copyWith();
-    _materialDarkTheme = _useMaterial3Dark == null
-        ? widget.materialDarkTheme
-        : widget.materialDarkTheme?.copyWith();
+    _materialLightTheme = widget.materialLightTheme;
+    _materialDarkTheme = widget.materialDarkTheme;
     _cupertinoLightTheme = widget.cupertinoLightTheme;
     _cupertinoDarkTheme = widget.cupertinoDarkTheme;
     _matchCupertinoSystemChromeBrightness =
         widget.matchCupertinoSystemChromeBrightness;
-
-    _useMaterial3Light =
-        _useMaterial3Light ?? widget.materialLightTheme?.useMaterial3;
-    _useMaterial3Dark =
-        _useMaterial3Dark ?? widget.materialDarkTheme?.useMaterial3;
   }
 
   @override
@@ -111,8 +102,6 @@ class _PlatformThemeState extends State<PlatformTheme>
         : themeMode == ThemeMode.dark;
   }
 
-  bool? _useMaterial3Light;
-  bool? _useMaterial3Dark;
   void changeToMaterial3({bool applyToBothDarkAndLightTheme = false}) =>
       _setMaterialThemeType(
           useMaterial3: true,
@@ -135,16 +124,12 @@ class _PlatformThemeState extends State<PlatformTheme>
   }) {
     setState(() {
       if (applyToBothDarkAndLightTheme) {
-        _useMaterial3Dark = _useMaterial3Light = useMaterial3;
-        _materialDarkTheme = _materialDarkTheme?.copyWith();
-        _materialLightTheme = _materialLightTheme?.copyWith();
+        _materialDarkTheme = _materialDarkTheme;
+        _materialLightTheme = _materialLightTheme;
       } else {
         isDark
-            ? _useMaterial3Dark = useMaterial3
-            : _useMaterial3Light = useMaterial3;
-        isDark
-            ? _materialDarkTheme = _materialDarkTheme?.copyWith()
-            : _materialLightTheme = _materialLightTheme?.copyWith();
+            ? _materialDarkTheme = _materialDarkTheme
+            : _materialLightTheme = _materialLightTheme;
       }
     });
   }

